#161351147 Add pagination support for getting articles from a category

emmaadesile opened the pull request:
### What does this PR do?
- add pagination to articles that belong to a category
- return a default of 6 articles per page for category articles
- return a default of 6 articles per page for favorite articles

### Description of tasks to be completed
- Add pagination support to articles gotten from a category

### How can this be manually tested
After cloning the branch,
- run `npm install` to install dependencies
- login as an author and get the token
- use token to create articles
- add the articles to a category at this endpoint`POST /api/article-categories/:categoryName/articles` 
- get the articles from that category at this endpoint ` GET /api/article-categories/:categoryName/articles`

### What are the relevant pivotal tracker stories?
[Deilvers 161351147]

emmaadesile committed:
ft(paginate-articles): paginate articles from a category

- add pagination to articles that belong to a category
- return a default of 6 articles per page for category articles
- return a default of 6 articles per page for favourite articles

[Deilvers 161351147]

d-beloved reviewed (approved):
LGTM

CEOehis commented on server/controllers/categoryController.js:
```diff
@@ -257,34 +259,66 @@ class CategoryController {
+                error: 'There are no articles in this category'
+              });
+            }
+            const pagination = paginateArticle(articles, currentPage, limit);
```
I don't think it should return an error if no articles exist for the category. It should return an empty array instead.

emmaadesile replied:
I hav fixed it

emmaadesile replied:
I have fixed it

emmaadesile replied:
Thanks for the feedback. I have fixed it. It now returns an empty array for categories without articles.

emmaadesile replied:
Thanks for the feedback. I have fixed it.

emmaadesile replied:
Thanks for the feedback. I have implemented the feedback

emmaadesile replied:
Thanks for the feedback. I have implemented the feedback.

emmaadesile replied:
Thanks for the feedback. I have implemented the feedback.

emmaadesile replied:
Thanks for the feedback. I have implemented the feedback.

emmaadesile replied:
Thanks for the feedback. I have implemented the feedback.

emmaadesile replied:
Thanks for the feedback. I have implemented the feedback.

emmaadesile committed:
nitpicks(pagination): add pagination for articles

- paginate articles drafts
- return a default of 12 article drafts per page
- return a default of 12 category articles per page
- return an empty array of there are no articles in a category

[Delivers 161351147]

CEOehis reviewed (approved):
👍LGTM

waleCloud commented on server/controllers/categoryController.js:
```diff
@@ -257,34 +259,60 @@ class CategoryController {
   static getAllArticlesForACategory(req, res, next) {
     const { categoryName } = req.params;
+    const { query } = req;
+    const limit = Number(query.limit) || 12;
```
Great job! Your commit messages say a limit of 6 articles but your code is showing 12. Are my missing something?

emmaadesile replied:
@waleCloud  Thanks for spotting that. That must have been an oversight. Initially, the code was returning 6 articles per page, but after getting feedback from my teammates, I changed it to 12.

emmaadesile replied:
@waleCloud I have squashed the commits. I now have 2 commit messages

waleCloud reviewed (changes requested):
Hello @emmaadesile, Can we follow Andela standard of commit message limits in PRs?
